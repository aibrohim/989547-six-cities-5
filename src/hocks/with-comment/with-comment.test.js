import React from "react";
import ShallowRenderer from 'react-test-renderer/shallow';
import propTypes from "prop-types";

import {withComment} from "./with-comment";

const noop = () => {};
const id = 0;

const MockComponent = (props) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

MockComponent.propTypes = {
  children: propTypes.oneOfType([
    propTypes.arrayOf(propTypes.node),
    propTypes.node
  ]).isRequired,
};

const MockComponentWrapped = withComment(MockComponent);

describe(`Should with comment work`, () => {
  it(`with no change comment sending status should render`, () => {
    const renderer = new ShallowRenderer();
    const tree = renderer
    .render(
        <MockComponentWrapped postCommentAction={noop} id={id}>
          <React.Fragment />
        </MockComponentWrapped>
    );

    expect(tree).toMatchSnapshot();
  });

  it(`with change comment sending status should render`, () => {
    const renderer = new ShallowRenderer();
    const tree = renderer
    .render(
        <MockComponentWrapped postCommentAction={noop} id={id} changeSendStatus={noop}>
          <React.Fragment />
        </MockComponentWrapped>
    );

    expect(tree).toMatchSnapshot();
  });
});
