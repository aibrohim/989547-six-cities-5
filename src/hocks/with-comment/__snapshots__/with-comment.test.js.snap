// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should with comment work with change comment sending status should render 1`] = `
<MockComponent
  buttonRef={
    Object {
      "current": null,
    }
  }
  changeSendStatus={[Function]}
  formRef={
    Object {
      "current": null,
    }
  }
  id={0}
  onFieldsChange={[Function]}
  onFormSubmit={[Function]}
  postCommentAction={[Function]}
>
  <React.Fragment />
</MockComponent>
`;

exports[`Should with comment work with no change comment sending status should render 1`] = `
<MockComponent
  buttonRef={
    Object {
      "current": null,
    }
  }
  formRef={
    Object {
      "current": null,
    }
  }
  id={0}
  onFieldsChange={[Function]}
  onFormSubmit={[Function]}
  postCommentAction={[Function]}
>
  <React.Fragment />
</MockComponent>
`;
